config/initializers/strong_migrations.rb

```diff
@@ -1,6 +1,6 @@
 # rubocop:disable Style/NumericLiterals
 unless Rails.env.production?
   StrongMigrations.start_after = 20200205123840
-  StrongMigrations.lock_timeout_limit = 86400.seconds
```

Comment:
This is a timeout of 24h, do we expect a migration to take longer?

Reply:
The problem with this check is, that it not actually checks how long a transaction takes but rather what `mariadb` is configured to (aka when to cancel a transaction).

See: https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_lock_wait_timeout which is fetched by the check here: https://github.com/ankane/strong_migrations/blob/master/lib/strong_migrations/adapters/mysql_adapter.rb#L35

To make this sane, we would need to ship our own mariadb configuration since the default is `31536000` and us setting `86400` will trigger this check and fail always with default configuration.

Reply:
Alright, I see. Why not remove that line entirely then instead of setting to 0? 

In glue we're setting timeouts for migrations, which I guess was the goal here too when we added that line: 

```
StrongMigrations.lock_timeout = 20.seconds
StrongMigrations.statement_timeout = 2.minutes
```

Would it make sense to set some limits for RMT too, so that a migration cannot block the db endlessly?

Reply:
I agree we should set the `lock_timeout` and `statement_timeout`, then also the `lock_timeout_limit` works. But without you will run into: https://github.com/ankane/strong_migrations/blob/master/lib/strong_migrations/checker.rb#L162 which then is setting by default: https://github.com/ankane/strong_migrations/blob/master/lib/strong_migrations.rb#L59

I'm not sure if setting a `statement_timeout` is really wanted, since a migration can take a long time which is completely fine.

Reply:
Ah, ok that's unexpected that strong migrations sets it's own default and fails then. Maybe a comment explaining this would help. 

Regarding the `statement_timeout`, it's possible to change it to a higher value for specific migrations that are known to take long.